src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidBrokerHelper.cs

```diff
@@ -208,31 +120,30 @@ public void CheckForBrokerAccountInfoInAccountManager(BrokerRequest brokerReques
             string homeAccountId = brokerRequest.HomeAccountId;
             string localAccountId = brokerRequest.LocalAccountId;
 
-            if (!string.IsNullOrEmpty(accounts))
+            if (!string.IsNullOrEmpty(accountData))
             {
-                dynamic authResult = JArray.Parse(accounts);
+                dynamic authResult = JArray.Parse(accountData);
 
                 foreach (JObject account in authResult)
                 {
-                    var accountData = account[BrokerResponseConst.Account];
-
-                    var accountDataHomeAccountID = accountData[BrokerResponseConst.HomeAccountId]?.ToString();
-                    var accountDataLocalAccountID = accountData[BrokerResponseConst.LocalAccountId]?.ToString();
+                    var accountInfo = account[BrokerResponseConst.Account];
+                    var accountInfoHomeAccountID = accountInfo[BrokerResponseConst.HomeAccountId]?.ToString();
+                    var accountInfoLocalAccountID = accountInfo[BrokerResponseConst.LocalAccountId]?.ToString();
 
-                    if (string.Equals(accountData[BrokerResponseConst.UserName].ToString(), username, StringComparison.OrdinalIgnoreCase))
+                    if (string.Equals(accountInfo[BrokerResponseConst.UserName].ToString(), username, StringComparison.OrdinalIgnoreCase))
                     {
                         // TODO: broker request should be immutable!
-                        brokerRequest.HomeAccountId = accountDataHomeAccountID;
-                        brokerRequest.LocalAccountId = accountDataLocalAccountID;
+                        brokerRequest.HomeAccountId = accountInfoHomeAccountID;
+                        brokerRequest.LocalAccountId = accountInfoLocalAccountID;
                         _logger.Info("Found broker account in Android account manager using the provided login hint. ");
-                        return;
+                        return brokerRequest;
                     }
 
-                    if (string.Equals(accountDataHomeAccountID, homeAccountId, StringComparison.Ordinal) &&
-                         string.Equals(accountDataLocalAccountID, localAccountId, StringComparison.Ordinal))
+                    if (string.Equals(accountInfoHomeAccountID, homeAccountId, StringComparison.Ordinal) &&
+                         string.Equals(accountInfoLocalAccountID, localAccountId, StringComparison.Ordinal))
```

Comment:
when would homeacount and localacocunt id be the same, and what does that mean? is there a reason to check for the user or just use home/local account?

Reply:
it may be possible that the user passes in a request with the localacocunt  id and the homeacount id. in that case we are just checking to see if the account data matches between the request data and the data in the account manager before executing the silent call